packages/rocketchat-livechat/client/views/app/tabbar/visitorInfo.js

```diff
@@ -22,6 +22,36 @@ Template.visitorInfo.helpers({
 		return !Template.instance().pageVisited.ready();
 	},
 
+	customFields() {
+		let fields = [];
+		let livechatData = {};
+		let user = Meteor.users.findOne({ 'profile.token': Template.instance().visitorToken.get() });
+		if (user) {
+			livechatData = _.extend(livechatData, user.livechatData);
+		}
+
+		let data = Template.currentData();
+		if (data && data.rid) {
+			let room = RocketChat.models.Rooms.findOne(data.rid);
+			if (room) {
+				livechatData = _.extend(livechatData, room.livechatData);
+			}
+		}
+
+		if (!_.isEmpty(livechatData)) {
+			for (let _id in livechatData) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2388610024/issues/source?bid=3199634&fileBranchId=3228833#l42)